mu-plugins/cdn/cdn-assets.php

```diff
@@ -0,0 +1,123 @@
+ * 
+ * In CDN'ing, there's a few specific changes made:
+ *  - ver= is always changed to the filemtime of the asset (if not already), for consistent and easy cache-busting.
+ *  - Caches are chunked into two-minute windows, to avoid slight time differences between servers using different cache keys.
+ *  - assets are shared between sites, all use s.w.org/* instead of wordpress.org/wp-includes/* or wordpress.org/plugins/wp-includes/*.
+ *  - non-production wp_get_envionment_type() skips CDN'isation.
+ * 
+ * @param string $link   The non-CDNised URL.
+ * @param string $handle The asset handle, used to skip certain assets.
+ * @return string The potentially CDNised URL.
+ */
+function with_filemtime_cachebuster( $link, $handle = '' ) {
+	// Only same-host resources, and WordPress.org domains.
+	if (
+		! str_starts_with( $link, site_url( '/' ) ) ||
+		! str_contains( $link, 'wordpress.org' ) ||
+		str_contains( $link, 'profiles.wordpress.org' )
+	) {
+		return $link;
+	}
+
+	$url_args     = [];
+	$relative_url = str_replace( site_url( '/' ), '', $link );
+
+	if ( str_contains( $relative_url, '?' ) ) {
+		list( $filepath, $url_part_args ) = explode( '?', $relative_url, 2 );
+
+		parse_str( $url_part_args, $url_args );
+	} else {
+		$filepath = $relative_url;
+		// No `$url_args` here.
+
+		// Webpack files often include the cache-buster in the filename, 'react' does this.
+		// Pretend that's the cache buster for the rest of the function.
+		if ( preg_match( '!\.([a-f0-9]{8})\.(min\.)?js$!', $filepath, $m ) ) {
+			$url_args = [
+				'ver' => $m[1]
+			];
+		}
+	}
+
+	// If the link doesn't have a cache-buster, or has extra args, abort.
+	if (
+		empty( $url_args['ver'] ) ||
+		count( $url_args ) > 1
+	) {
+		return $link;
+	}
+
+	// Set the version to the file modification time, for consistency.
+	$version = false;
+	if ( ! is_timestamp( $url_args['ver'] ) ) {
+		$version = filemtime( ABSPATH . $filepath );
+	}
+	if ( ! $version ) {
+		$version = $url_args['ver'];
+	}
+
+	// Chunk the cache buster on a ~2 minute rolling window.
+	// This allows for the production deploy process taking a few minutes, setting different modification times on different servers.
+	if (
+		'production' === wp_get_environment_type() &&
+		is_timestamp( $version )
+	) {
+		$window  = 2 * MINUTE_IN_SECONDS;
+		$version = floor( $version / $window ) * $window;
+	}
+
+	$use_cdn = (
+		// CDN is used in production by default.
+		'production' === wp_get_environment_type() ||
+
+		// Allow other environments to opt-in via constant.
+		( defined( 'USE_WPORG_CDN' ) && USE_WPORG_CDN )
+	);
+
+	// These scripts/styles break if loaded from the CDN without CORS headers.
+	// See https://make.wordpress.org/systems/2022/02/22/cors-headers-for-s-w-org/
+	$blocked_handles = [
+		'wporg-news-style', // Mask images
+		'wporg-global-fonts-css', 'jetpack-icons', 'wp-stream-icons', // Fonts
+	];
+	if ( in_array( $handle, $blocked_handles, true ) ) {
+		$use_cdn = false;
+	}
```

Comment:
```suggestion
```

Remove the block list, as https://make.wordpress.org/systems/2022/02/22/cors-headers-for-s-w-org/ results in the mask images and fonts serving correctly.